### PR TITLE
fix a websocket origin issue

### PR DESCRIPTION
**Pull request by devoncarew**
- fix an issue with debug protocol changes - use a node WebSocket implementation instead of the browser based one (it was reporting `file://` as the socket origin)
- improve an error message when launching Flutter apps
- allow more than one Flutter app to run at a time
- when re-running an already running app, choose to restart the app if the target supports it

@danrubel

### CODE REVIEW

File: lib/flutter/flutter_launch.dart [hunk 1]
```diff
@@ -68,7 +70,27 @@ class FlutterLaunchType extends LaunchType {
         "settings for the 'flutter' plugin and / or try re-starting Atom.";
     }
 
-    await _killLastLaunch();
+    // Instead of killing the last launch, check if we should re-start it.
+    if (_lastLaunch != null) {
```
Comment (danrubel) [inline lib/flutter/flutter_launch.dart]: Combine the two `if (_lastLaunch != null) {` statements ?

File: lib/debug/websocket.dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import 'dart:async';
+import 'dart:js';
+
+import 'package:atom/node/node.dart';
+
+class WebSocket {
```
Comment (danrubel) [inline lib/debug/websocket.dart]: At some point in the future, should we move this class into the `atom.dart` plugin?

**Comment by danrubel** [comment]
LGTM

File: lib/debug/websocket.dart (hunk 2 above)
Comment (devoncarew) [inline lib/debug/websocket.dart]: Yup, will add a TODO:

File: lib/flutter/flutter_launch.dart (hunk 1 above)
Comment (devoncarew) [inline lib/flutter/flutter_launch.dart]: done

**Comment by devoncarew** [comment]
Thanks for the review! Landing...

**Pull request merged**